internal/index/indexoperations/index_test.go

```diff
@@ -64,3 +54,65 @@ func TestListIndexes(t *testing.T) {
 		t.Errorf("output does not match: %s", diff)
 	}
 }
+
+func TestAddIndex(t *testing.T) {
+	type args struct {
+		index Index
+	}
+	tests := []struct {
+		name    string
+		args    args
+		wantErr bool
+	}{
+		{
+			name: "invalid git repository",
+			args: args{
+				index: Index{
+					Name: "foo",
+				},
+			},
+			wantErr: true,
+		},
+		{
+			name: "preexisting index",
+			args: args{
+				index: Index{
+					Name: "foo",
+					URL:  "https://github.com/valid/index.git",
+				},
+			},
+			wantErr: true,
+		},
+	}
```

Comment:
Add success case

Reply:
Is it ok for it to clone a repo during the unit tests? I had a success case initially using krew-index but didn't like that it required a network connection.

Reply:
You can clone local dirs as repos IIRC

Reply:
https://stackoverflow.com/questions/21045061/git-clone-from-another-directory

Reply:
Maybe you can make success test another unit test method. Tabular tests are often overkill for 1-2 cases.